x-pack/plugins/maps/public/classes/sources/ems_tms_source/ems_tms_source.js

```diff
@@ -55,17 +65,30 @@ export class EMSTMSSource extends AbstractTMSSource {
         value: this._descriptor.isAutoSelect ? `${displayName} - ${autoSelectMsg}` : displayName,
       },
     ];
+
+    const emsSettings = getEMSSettings();
+    if (emsSettings.isEMSUrlSet()) {
+      props.push({
+        label: i18n.translate('xpack.maps.source.emsTile.emsOnPremLabel', {
+          defaultMessage: `Elastic Maps Server`,
+        }),
+        value: emsSettings.getEMSRoot(),
+      });
+    }
+
+    return props;
   }
 
   async _getEMSTMSService() {
     const emsTMSServices = await getEmsTmsServices();
     const emsTileLayerId = this.getTileLayerId();
     const tmsService = emsTMSServices.find((tmsService) => tmsService.getId() === emsTileLayerId);
     if (!tmsService) {
+      const info = getEmsUnavailableMessage();
```

Comment:
Aside from disabling the endpoint in Elastic Maps Server, I can't see a way to reach this error. If Elastic Maps Server is unreachable, Kibana will fail first to request the manifest and report a Fetch error. Is it possible to add the `getEmsUnavailableMessage` function to the request for the manifest? Or is that abstracted away by the ems-client?

Reply:
Thx, good point. Now it only captured if EMS was unavailable through known error. Now, it explicitly captures any internal errors (e.g. from EMSclient) and makes them part of the error message.